spec/integration/blog_app_intergration_spec.rb

```diff
@@ -0,0 +1,158 @@
+    @post2 = Post.create(title: 'title 2',
+                         text: 'post content 2',
+                         comments_counter: 0,
+                         likes_counter: 0,
+                         author_id: @jin.id)
+    @post3 = Post.create(title: 'title 3',
+                         text: 'post content 3',
+                         comments_counter: 0,
+                         likes_counter: 0,
+                         author_id: @jin.id)
+    @post4 = Post.create(title: 'title 4',
+                         text: 'post content 4',
+                         comments_counter: 0,
+                         likes_counter: 0,
+                         author_id: @jin.id)
+    @comment1 = Comment.create(author_id: @profile1.id, post_id: @post1.id, text: 'Hello')
+    @comment2 = Comment.create(author_id: @profile1.id, post_id: @post1.id, text: 'wassup')
+    @like1 = Like.create(author_id: @profile1.id, post_id: @post1.id)
+  end
+  describe 'user index' do
+    before(:each) do
+      visit users_path
+    end
+    it 'Display the user details' do
+      expect(page).to have_content(@profile1.name)
+      expect(page).to have_content(@profile2.name)
+    end
+
+    it 'Display each user profile picture' do
+      expect(page).to have_xpath("//img[contains(@src,'https://as2.ftcdn.net/v2/jpg/03/26/98/51/1000_F_326985142_1aaKcEjMQW6ULp6oI9MYuv8lN9f8sFmj.jpg')]")
+    end
+
+    it 'Total number of posts' do
+      expect(page).to have_content('Total number of posts: 0')
+    end
+
+    it 'Redirect to user page' do
+      click_link 'Nabeel'
+      expect(page).to have_current_path("/users/#{@profile1.id}")
+    end
+  end
+
+  describe 'Show user' do
+    before(:each) do
+      visit user_path(@jin.id)
+    end
+
+    it 'Display each user profile picture' do
+      expect(page).to have_xpath("//img[contains(@src,'https://as2.ftcdn.net/v2/jpg/03/26/98/51/1000_F_326985142_1aaKcEjMQW6ULp6oI9MYuv8lN9f8sFmj.jpg')]")
+    end
+    it 'Display each user name' do
+      expect(page).to have_content(@jin.name)
+      expect(page).to_not have_content(@profile2.name)
+    end
+    it 'Should show Total number of posts by each user' do
+      expect(page).to have_content 'Total number of posts: 4'
+    end
+    it 'Should display user bio' do
+      expect(page).to have_content 'Teacher from Mexico.'
+    end
+    it 'I can see the user recent 3 posts.' do
+      expect(page).to_not have_content 'post content 4'
+    end
+    it 'I can see a button that show user all posts' do
+      expect(page).to have_content 'See all posts'
+    end
+    it 'When I click a users post, it redirects me to that posts show page.' do
+      visit "/users/#{@jin.id}/posts/#{@post1.id}"
+      expect(page).to have_content 'post content 1'
+    end
+    it 'When I click to see all posts, it redirects me to the users posts index page.' do
+      click_link 'See all posts'
+      expect(page).to have_content 'title 1'
+    end
+  end
+
+  describe 'post#index' do
```

Comment:
- [Optional] wonderful job so far. Please always take note of the requirements, so your work can be more professional. Nice work correcting and fixing the test. Please, all the test were to be in difference spec files. So you needed separate files for
- `User post show page,`, `User post index page`.,, `User show page:` and `user index page`
since the previous reviewer forgot to mention this, kindly fix this, so your work can be superb. Nice work.

Reply:
Thank you so much

I shall make the change. I got late due to webdriver issue and need to complete one more project now.